src/components/NewSnippet.jsx

```diff
@@ -1,49 +1,87 @@
 import React from 'react';
+import { connect } from 'react-redux';
 import { Controlled as CodeMirror } from 'react-codemirror2';
 
 import 'codemirror/lib/codemirror.css';
 
 import Title from './common/Title';
 import Input from './common/Input';
 import Syntaxes from './Syntaxes';
+import * as actions from '../actions';
 
 import '../styles/NewSnippet.styl';
 
 class NewSnippet extends React.Component {
-  constructor() {
-    super();
-    this.state = { code: '' };
+  constructor(props) {
+    super(props);
+    this.state = {
+      content: '',
+      title: '',
+      tags: [],
+      syntax: '', // eslint-disable-line react/no-unused-state
+    };
+    this.postSnippet = this.postSnippet.bind(this);
+    this.onSyntaxClick = this.onSyntaxClick.bind(this);
+    this.onInputChange = this.onInputChange.bind(this);
   }
+
+  onSyntaxClick(syntax) {
+    this.setState({ syntax }); // eslint-disable-line react/no-unused-state
+  }
+
+  onInputChange(e) {
+    const { name, value } = e.target;
+    const result = (name === 'tags')
+      ? value.split(',').map(item => item.trim())
+      : value;
+
+    this.setState({ [name]: result });
```

Comment:
Why `name` is wrapped in brackets (i.e. `[name]`)?

Reply:
I want to reuse this function for a few similar inputs, and I want to set their values in state, so I will have: `{tags: some_value}` and `{title: some_other_value}`.
If I will use name without [] it will think that I want this `{name: some_value}`.
In this case js syntax lets me to use name as different key

Reply:
Oh, gosh, Jesus.. so basically if you use `name` without brackets, it will think that `name` is a name of key?

Reply:
yeap, exactly

Reply:
🤦‍♂️ what a silly language